controllers/tc000015_source_cross_ns_denied_test.go

```diff
@@ -0,0 +1,111 @@
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+	infrav1 "github.com/weaveworks/tf-controller/api/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+func Test_000015_cross_namespace_source_denied_test(t *testing.T) {
+	Spec("This spec describes the behaviour of a Terraform resource when source is in another namespace and corss-namespace refs are not allowed.")
+	It("should be reconciled to have a Source error.")
+
+	By("setting the reconciler to disallow cross-namespace refs")
+	defer func(original bool) {
+		reconciler.NoCrossNamespaceRefs = original
+	}(reconciler.NoCrossNamespaceRefs)
+	reconciler.NoCrossNamespaceRefs = true
+
+	const (
+		sourceName    = "gr-source"
+		terraformName = "tf-cross-ns"
+	)
+	g := NewWithT(t)
+	ctx := context.Background()
+
+	Given("a GitRepository")
+	By("defining a new GitRepository resource.")
+	testRepo := sourcev1.GitRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourceName,
+			Namespace: "flux-system",
+		},
+		Spec: sourcev1.GitRepositorySpec{
+			URL: "https://github.com/openshift-fluxv2-poc/podinfo",
+			Reference: &sourcev1.GitRepositoryRef{
+				Branch: "master",
+			},
+			Interval: metav1.Duration{Duration: time.Second * 30},
+		},
+	}
+	By("creating the GitRepository resource in the cluster.")
+	It("should be created successfully.")
+	g.Expect(k8sClient.Create(ctx, &testRepo)).Should(Succeed())
+	defer func() { g.Expect(k8sClient.Delete(ctx, &testRepo)).Should(Succeed()) }()
+
+	Given("a Terraform resource, attached to a GitRepository resource in another namespace.")
+	By("creating a new TF resource and attaching to the repo via `sourceRef`.")
+	helloWorldTF := infrav1.Terraform{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      terraformName,
+			Namespace: "default",
+		},
+		Spec: infrav1.TerraformSpec{
+			ApprovePlan: "auto",
+			Path:        "./terraform-hello-world-example",
+			SourceRef: infrav1.CrossNamespaceSourceReference{
+				Kind:      "GitRepository",
+				Name:      sourceName,
+				Namespace: "flux-system",
+			},
+			Interval: metav1.Duration{Duration: time.Second * 10},
+		},
+	}
+	It("should be created and attached successfully.")
+	g.Expect(k8sClient.Create(ctx, &helloWorldTF)).Should(Succeed())
+	defer func() { g.Expect(k8sClient.Delete(ctx, &helloWorldTF)).Should(Succeed()) }()
+
+	It("should be reconciled and contain some status conditions.")
+	By("checking that the TF resource's status conditions has some elements.")
+	helloWorldTFKey := types.NamespacedName{Namespace: "default", Name: terraformName}
+	var createdHelloWorldTF infrav1.Terraform
+	g.Eventually(func() int {
+		err := k8sClient.Get(ctx, helloWorldTFKey, &createdHelloWorldTF)
+		if err != nil {
+			return -1
+		}
+		return len(createdHelloWorldTF.Status.Conditions)
+	}, timeout, interval).ShouldNot(BeZero())
```

Comment:
Could this be better expressed as expecting the `.Get` to `Succeed()` and returning `createdHelloWorldTF.Status.Conditions` and waiting for it not to `BeEmpty()` ?

Reply:
The code following fetches the object and checks for a ready condition, which obviates this check, so I removed it.